lib/services/twitter.rb

```diff
@@ -42,9 +42,9 @@ def receive_push
         url = commit['url']
         # Strip out leading @s so that github @ mentions don't become twitter @ mentions
         # since there's zero reason to believe IDs on one side match IDs on the other
-        message = commit['message'].split(' ').map do |word|
-          (word.length > 1 && word[0] == '@') ? "@\u200b#{word[1..word.length]}" : word
-        end.join(' ')
+        message = commit['message'].gsub(/\B@[^ ]+/) do |word|
```

Comment:
You're right, `[^ ]` is definitely too specific. I looked at what Twitter themselves look for when matching mentions, and it looks like the `@` [must be followed by `[a-zA-Z0-9_]{1,20}`][tt].

I changed the regex to use `[^[:word:]]` and added tests for this behavior.

[tt]: https://github.com/twitter/twitter-text/blob/master/rb/lib/twitter-text/regex.rb#L122